qa-pipelines/tasks/cf-teardown.sh

```diff
@@ -3,16 +3,22 @@ set -o errexit -o nounset
 
 # Set kube config from pool
 source "ci/qa-pipelines/tasks/lib/prepare-kubeconfig.sh"
+source "ci/qa-pipelines/tasks/lib/klog-collection.sh"
+
 set -o allexport
 CF_NAMESPACE=scf
 UAA_NAMESPACE=uaa
 set +o allexport
 
-for namespace in "$CF_NAMESPACE" "$UAA_NAMESPACE" ; do
+set -- "${CF_NAMESPACE}" "${UAA_NAMESPACE}"
+while [[ -n "${1}" ]]; do
+    # Upload klogs for any releases which have not yet been successfully deleted
+    trap "upload_klogs_on_failure ${*}" EXIT
     while [[ -n $(helm list --short --all ${namespace}) ]]; do
         helm delete --purge ${namespace} ||:
         sleep 10
     done
+    shift
 done
```

Comment:
Shouldn't you `trap "" EXIT` there? There's no point getting logs anymore after you have finished deleting things.

Reply:
At the start of the `cf-teardown` we trap EXIT with `upload_klogs_on_failure cf uaa`. cf gets deleted first (if a failure happens on deletion here, we want klogs for both), and after deletion, positional params shift, and then EXIT is trapped with `upload_klogs_on_failure uaa` before uaa deletion is attempted. After `done`, it would make sense to clear EXIT trap command though

Reply:
Err, yes, exactly? At the end of the loop, it's kind of pointless to look for logs of the things that have already been deleted. That's why I made the comment on the `done` line :)